include/CL/sycl/buffer.hpp

```diff
@@ -211,27 +212,53 @@ class buffer : public detail::property_carrying_object
       // the memory range [first, last) as writeback buffer
       this->_buffer->enable_write_back(true);
 
-      
+
       auto buff = this->_buffer;
       this->_cleanup_trigger->add_cleanup_callback([first, last, buff](){
         const T* host_data = reinterpret_cast<T*>(buff->get_host_ptr());
 
         std::copy(host_data, host_data + std::distance(first,last), first);
       });
-      
+
     }
   }
 
   template <class InputIterator, int D = dimensions,
             typename = std::enable_if_t<D == 1>>
   buffer(InputIterator first, InputIterator last,
          const property_list &propList = {})
-  : buffer(first, last, AllocatorT(), propList) 
+  : buffer(first, last, AllocatorT(), propList)
   {}
 
-  buffer(buffer<T, dimensions, AllocatorT> b,
-         const id<dimensions> &baseIndex,
-         const range<dimensions> &subRange);
+  buffer(buffer<T, dimensions, AllocatorT> &b,
+          const id<dimensions> &baseIndex,
+          const range<dimensions> &subRange,
+          const property_list &propList = {})
+      : detail::property_carrying_object{propList},
+      _range(subRange), _is_subbuf{true}
+  {
+      std::cerr << "Sub buffer status \n" << is_sub_buffer() << " \n"<< std::endl << std::flush;
+
+      auto b_ = b._detail_get_buffer_ptr();
+      size_t offset = baseIndex.get(0);
+      size_t size = subRange.size() * sizeof(T);
+
+      // Let's do some void* pointer arithmetic!
+      T* buff = static_cast<T*>(b_->get_buffer_ptr());
+      T* host = static_cast<T*>(b_->get_host_ptr());
+
+      std::cerr << "Status ok" << std::endl << std::flush;
+
+      _buffer = detail::buffer_ptr{
+        new detail::buffer_impl{
+          (void*)(buff + offset),
```

Comment:
What happens if `dimensions > 1`?

Reply:
This is just a stopgap meant to support d=1